packages/contracts/test/test-utils/uups-upgradeable.ts

```diff
@@ -1,96 +1,103 @@
 import {SignerWithAddress} from '@nomiclabs/hardhat-ethers/signers';
 import {expect} from 'chai';
-import {Contract} from 'ethers';
-import {defaultAbiCoder} from 'ethers/lib/utils';
-import {ethers} from 'hardhat';
-import {PluginUUPSUpgradeableV1Mock__factory} from '../../typechain';
+import {ContractFactory} from 'ethers';
+import {upgrades} from 'hardhat';
+import {DAO} from '../../typechain';
 
 // See https://eips.ethereum.org/EIPS/eip-1967
 export const IMPLEMENTATION_SLOT =
   '0x360894a13ba1a3210667c828492db98dca3e2076cc3735a920a3ca505d382bbc'; // bytes32(uint256(keccak256('eip1967.proxy.implementation')) - 1)
 
-/// Used as a common test suite to test upgradeability of the contracts.
-/// Presumes that `upgrade` object is set on `this` inside the actual test file.
-/// this.upgrade consists of:
-///     contract - address of the contract on which it tests if `upgradeTo` works as intended.
-///     dao - dao contact that the contract belongs to.
-///     user - ethers user object. Presumed that it doesn't have permission to call `upgradeTo`.
-export function shouldUpgradeCorrectly(
-  upgradePermissionId: string,
-  upgradeRevertPermissionMessage: string
+// Deploys and upgrades a contract that is managed by a DAO
+export async function upgradeCheckManagedContract(
+  deployer: SignerWithAddress,
+  upgrader: SignerWithAddress,
+  managingDao: DAO,
+  initArgs: any,
+  initializerName: string,
+  from: ContractFactory,
+  to: ContractFactory,
+  upgradePermissionId: string
 ) {
```

Comment:
don't we need to return the upgrade tx, and implementation...etc?
so the consumer of this function could do addition test after the upgrade like verify implementation, verify initialized version...etc